string name
string[] running_components

